src/admin/admin.controller.ts

```diff
@@ -1,19 +1,20 @@
-import { Controller, Post, Body, Patch, Param, Get } from '@nestjs/common';
+import { Controller, Body, Patch } from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { AdminService } from './admin.service';
+import { SuspendOrActivateDTO } from './admin.dto';
 
 @ApiTags('Admin')
 @Controller('admin')
 export class AdminController {
     constructor(private readonly service: AdminService) {}
 
     @Patch('suspend')
-    suspend(@Body('id') id: string) {
-        return this.service.suspend(id);
+    suspend(@Body() { userId }: SuspendOrActivateDTO) {
+        return this.service.suspend(userId);
     }
 
     @Patch('activate')
-    activate(@Body('id') id: string) {
-        return this.service.activate(id);
+    activate(@Body() { userId }: SuspendOrActivateDTO) {
+        return this.service.activate(userId);
```

Comment:
what does DTO help here? Is it about and option in API Prop? And if it is, what will happen after request Body not matching with DTO?

Reply:
Right now it has no effect, but making it as a class is helpful when validate data.